scanpipe/pipelines/populate_purldb.py

```diff
@@ -48,6 +50,18 @@ def populate_purldb_with_discovered_dependencies(self):
             package_type="DiscoveredDependency",
         )
 
+    def populate_purldb_with_detected_purls(self):
+        """Add DiscoveredPackage to PurlDB."""
+        if (
+            not self.project.discoveredpackages.exists()
+            and not self.project.discovereddependencies.exists()
+        ):
```

Comment:
You can wrap the condition into a explicit variable to help the readability.
```suggestion
        no_packages_and_no_dependencies = all([
            not self.project.discoveredpackages.exists(),
            not self.project.discovereddependencies.exists(),
        ])
        
        if no_packages_and_no_dependencies:
```

Reply:
Added this, thanks!